modin/pandas/indexing.py

```diff
@@ -821,12 +821,12 @@ def _setitem_with_new_columns(self, row_loc, col_loc, item):
         if is_list_like(item) and not isinstance(item, (DataFrame, Series)):
             item = np.array(item)
             if len(item.shape) == 1:
-                if item.shape[0] != len(col_loc):
+                if len(col_loc) != 1:
```

Comment:
Hi @JignyasAnand! Thanks for the fix! 

Shouldn't we compare the item length with `len(row_loc)` here too?

Reply:
Got it, thanks for the explanation!

> So, trying to get len(row_loc) always results in TypeError: object of type 'slice' has no len()

I assume the same problem is present for `col_loc` indexer? (If so, could you create a new issue, not necessarily fix it here)

Reply:
Here, the `col_loc` is a list. So, there is no problem with `col_loc`

Reply:
If so could you please update docstring of the method:
`col_loc : scalar, slice, list, array or tuple` -> `col_loc : list, array or tuple`

Reply:
Done